src/reducers/computeRenderedColumns.js

```diff
@@ -139,6 +139,88 @@ function calculateRenderedColumnRange(state, columnAnchor) {
   };
 }
 
+/**
+ * Determine the range amd offsets of column groups to be rendered (buffer and viewport)
+ *
+ * NOTE (jordan) This alters state so it shouldn't be called
+ * without state having been cloned first.
+ *
+ * @param {!Object} state
+ * @param {{
+ *   firstIndex: number,
+ *   firstOffset: number,
+ *   lastIndex: number,
+ * }} columnAnchor
+ * @param {{
+ *   endBufferCol: number,
+ *   endViewportCol: number,
+ *   firstBufferCol: number,
+ *   firstViewportCol: number,
+ * }} columnRange
+ * @private
+ */
+function calculateRenderedColumnGroups(state, columnAnchor, columnRange) {
+  const bufferColumnCount = 0; // TODO (pradeep): calculate this similar to bufferRowCount
+  const { columnGroupBufferSet, columnOffsetIntervalTree } = state;
+
+  const { availableScrollWidth, scrollableColumns, scrollableColumnGroups } = columnWidths(state);
+  const columnCount = scrollableColumns.length;
+  const columnGroupCount = scrollableColumnGroups.length;
+
+  if (columnCount === 0 || columnGroupCount === 0) {
+    state.firstColumnGroupIndex = 0;
+    state.endColumnGroupIndex = 0;
+    state.columnGroupsToRender = [];
+    state.columnGroupOffsets = {};
+    return;
+  }
+
+  // get first and last scrollable columns in the view port
+  const startScrollableColumn = scrollableColumns[Math.max(0, columnRange.firstViewportCol)];
+  const endScrollableColumn = scrollableColumns[Math.max(0, columnRange.endViewportCol - 1)];
+
+  // now get the first and last scrollable group column's index in the view port
+  const startIdx = startScrollableColumn.parentIdx;
+  const endIdx = endScrollableColumn.parentIdx + 1;
```

Comment:
Why is the index for the endScrollableColumn different? I see columnRange.endViewportCol - 1, then this line. Why is there this discrepancy in endScrollableColumn, but not startScrollableColumn?